validator/src/main/java/com/amazon/aoc/validators/ContainerInsightMetricsValidator.java

```diff
@@ -0,0 +1,76 @@
+package com.amazon.aoc.validators;
+
+import com.amazon.aoc.callers.ICaller;
+import com.amazon.aoc.exception.BaseException;
+import com.amazon.aoc.exception.ExceptionCode;
+import com.amazon.aoc.fileconfigs.FileConfig;
+import com.amazon.aoc.helpers.MustacheHelper;
+import com.amazon.aoc.helpers.RetryHelper;
+import com.amazon.aoc.models.Context;
+import com.amazon.aoc.models.ValidationConfig;
+import com.amazon.aoc.services.CloudWatchService;
+import com.amazonaws.services.cloudwatch.model.Metric;
+import com.amazonaws.services.cloudwatch.model.MetricDataResult;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import lombok.extern.log4j.Log4j2;
+
+import java.nio.charset.StandardCharsets;
+import java.sql.Date;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+@Log4j2
+public class ContainerInsightMetricsValidator implements IValidator {
+  private CloudWatchService cloudWatchService;
+  private List<Metric> expectedMetrics;
+  ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
+
+  private static final int MAX_RETRY_COUNT = 6;
+  private static final int CHECK_INTERVAL_IN_MILLI = 30 * 1000;
+  private static final int CHECK_DURATION_IN_SECONDS = 2 * 60;
+
+  @Override
+  public void init(Context context, ValidationConfig validationConfig, ICaller caller,
+                   FileConfig expectedDataTemplate) throws Exception {
+    cloudWatchService = new CloudWatchService(context.getRegion());
+    MustacheHelper mustacheHelper = new MustacheHelper();
+    String templateInput = mustacheHelper.render(expectedDataTemplate, context);
+    expectedMetrics = mapper.readValue(templateInput.getBytes(StandardCharsets.UTF_8),
+          new TypeReference<List<Metric>>() {
+          });
+  }
+
+  @Override
+  public void validate() throws Exception {
```

Comment:
what's the difference between appmesh validation and the normal cloudwatch metric validation? seems we just do a simple check on it? is it intended or we had a "todo" for it?

Reply:
Yes, I guess we need a discussion on how to merge the two validators. There're some assumption that breaks container insight validations in the original cw metric validator.

Reply:
so, what's the path forward now? we're gonna use this simple validation instead of making the validator more generic? i'd prefer we do the latter, but if we have to do so, let's add a "todo" here

Reply:
I prefer we add todo and deal with it later, this PR is already pretty big and we are not enabling it until the related features merge into adot repo. We have plenty time to refactor those validators, both me and @pxaws will use cw metrics and log validators for other container insight related features.